application/F3DStarter.cxx

```diff
@@ -330,10 +332,33 @@ void F3DStarter::LoadFile(f3d::loader::LoadFileEnum load)
     }
   }
 
-  // Recover info about the file to be loaded
-  int index;
+  // Recover file information
   std::string filePath, fileName, fileInfo;
-  this->Internals->Engine->getLoader().getFileInfo(load, index, filePath, fileName, fileInfo);
+  int size = static_cast<int>(this->Internals->FilesList.size());
+  if (size > 0)
+  {
+    if (load == LoadFileEnum::LOAD_LAST)
+    {
+      this->Internals->CurrentFileIndex = size - 1;
+    }
+    else
+    {
+      this->Internals->CurrentFileIndex += load;
+      this->Internals->CurrentFileIndex %= size;
+      this->Internals->CurrentFileIndex = this->Internals->CurrentFileIndex < 0
+        ? this->Internals->CurrentFileIndex + size
+        : this->Internals->CurrentFileIndex;
```

Comment:
Would it be worth defining a proper modulo function? to do something like:
```cpp
switch(load){
  case LOAD_FIRST: i = 0; break
  case LOAD_LAST: i = size-1; break
  case LOAD_NEXT: i = modulo(i+1, size); break
  case LOAD_PREV: i = modulo(i-1, size); break
}```

Reply:
Definitely, for now I'm just moving methods around before the actual redesign

Reply:
reworked the whole thing